lib/puppet/pops/lookup/data_provider.rb

```diff
@@ -0,0 +1,84 @@
+module Puppet::Pops
+module Lookup
+# @api private
+module DataProvider
+  # The Pcore type for all keys and subkeys in a data hash.
+  TYPE_DATA_KEY = Types::PVariantType.new([
+    Types::PStringType::NON_EMPTY,
+    Types::PBooleanType::DEFAULT,
```

Comment:
This is a bit surprising - is this a legacy from hiera (I think it means you can bind values to the boolean keys true and false)?